### ISSUE TITLE
Improved linker defaults

### ISSUE THREAD

**Issue by SteveSandersonMS** [aspnet/Blazor#1316]
- [x] Consider Mono team's recommendations for a linker config that can strip out more. **Done**
- [x] ~~Make sure Json.NET works out of the box~~ See below
- [ ] Consider changing default build setup so that we only strip the BCL, not any other referenced assemblies

**Comment by SteveSandersonMS** [comment]
Update: We've heard from the Mono team that, in an update coming soon, they are changing how LINQ expressions are handled internally. The change will mean Json.NET will work out of the box without linker config changes, and the BCL will get a bit smaller, and the LINQ expressions handling will be substantially faster.

**Comment by danroth27** [comment]
We expect the update will come post 0.6.0 though.

**Comment by SteveSandersonMS** [comment]
Oh yeah sorry, should have clarified that.

**Comment by SteveSandersonMS** [comment]
Another update: while experimenting, I found that because our existing linker build logic passes *all* of the user assembly paths to the linker explicitly (via `-a assembly.dll`), in effect we already don't link out code from user assemblies. If we do want to link out code from user assemblies, we need to change this to `-d <dir_containing_assembly>` for each user assembly other than the entrypoint.

I tried changing it so it can strip code from user assemblies, and it only reduces the default app payload by 30KB. This does not seem like a worthwhile gain, considering the additional risk.
